### PR TITLE
6 retrieval

### PR DESCRIPTION
**Pull request by Sunalwing**

### CODE REVIEW

File: mteb/tasks/Retrieval/AlloprofRetrieval.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            "description": (
+                "This dataset was provided by AlloProf, an organisation in Quebec, Canada offering resources and a help forum"
+                "curated by a large number of teachers to students on all subjects taught from in primary and secondary school"
+            ),
+            "type": "Retrieval",
+            "category": "s2p",
+            "eval_splits": ["test"],
+            "eval_langs": ["fr"],
+            "main_score": "ndcg_at_10",
+        }
```
Comment (GabrielSequeira) [inline mteb/tasks/Retrieval/AlloprofRetrieval.py:23]: missing revision tag

File: mteb/tasks/Retrieval/BSARDRetrieval.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                "is a French native dataset for studying legal information retrieval."
+                "BSARD consists of more than 22,600 statutory articles from Belgian law"
+                "and about 1,100 legal questions posed by Belgian citizens and labeled"
+                "by experienced jurists with relevant articles from the corpus."
+            ),
+            "type": "Retrieval",
+            "category": "s2p",
+            "eval_splits": ["test"],
+            "eval_langs": ["fr"],
+            "main_score": "ndcg_at_10",
```
Comment (GabrielSequeira) [inline mteb/tasks/Retrieval/BSARDRetrieval.py:27]: missing revision tag

File: mteb/tasks/Retrieval/HagridRetrieval.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            "description": (
+                "HAGRID (Human-in-the-loop Attributable Generative Retrieval for Information-seeking Dataset)"
+                "is a dataset for generative information-seeking scenarios. It consists of queries"
+                "along with a set of manually labelled relevant passages"
+            ),
+            "type": "Retrieval",
+            "category": "s2p",
+            "eval_splits": ["test"],
+            "eval_langs": ["en"],
+            "main_score": "ndcg_at_10",
```
Comment (GabrielSequeira) [inline mteb/tasks/Retrieval/HagridRetrieval.py:27]: missing revision tag

File: mteb/tasks/Retrieval/HagridRetrieval.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+    def load_data(self, **kwargs):
+        """
+        Loads the different split of the dataset (queries/corpus/relevants)
+        """
+        if self.data_loaded:
+            return
+
+        data = datasets.load_dataset("miracl/hagrid", split="dev")
```
Comment (GabrielSequeira) [inline mteb/tasks/Retrieval/HagridRetrieval.py:37]: for split use the parameter instead of "dev"

File: mteb/tasks/Retrieval/__init__.py [hunk 5]
```diff
@@ -37,4 +37,7 @@
 from .SCIDOCSPLRetrieval import *
 from .SciFactPLRetrieval import *
 from .TRECCOVIDPLRetrieval import *
-from .NarrativeQARetrieval import *
\ No newline at end of file
+from .NarrativeQARetrieval import *
+from .AlloprofRetrieval import *
+from .BSARDRetrieval import *
+from .HagridRetrieval import *
```
Comment (GabrielSequeira) [inline mteb/tasks/Retrieval/__init__.py:43]: Alphabetic order like the others files

**Review by GabrielSequeira — CHANGES_REQUESTED** [review]

**Review by imenelydiaker — APPROVED** [review]

**Pull request merged**